### ISSUE TITLE
Allow resubmission in exam mode

### ISSUE THREAD

**Issue by rzach** [Carnap/Carnap#129]
Carnap does not allow resubmission of a problem once it has been submitted. In exam mode this means a student has to decide whether to wait until the deadline to submit, or risk submitting before they have the best solution they can find. This makes for a stressful exam experience. It could be alleviated simply by allowing solutions to be overridden in exam mode, perhaps with a warning "You have already submitted a solution to this problem. Click OK to replace it."
(Perhaps can be combined with #91 )

**Comment by gleachkr** [comment]
Good idea. This should probably be coordinated with a change that hides exam scores until a certain time.

**Comment by gleachkr** [comment]
Also - in connection with #154, and some ideas for problem-chaining, it might make sense to make this part of a more general feature that allows replacing one problem submission with having given label with another one at the same label.
